File: DesignPatternsExercises/ruby/strategy_design_pattern/solution/strategy_design_pattern.rb
require_relative 'csv_data_type'
require_relative 'db_data_type'
require_relative 'tsv_data_type'

puts 'Extraction tool for CSV file:'
data = CSVDataType.new
data.extract_data

puts "\nExtraction tool for DB file:"
data = DBDataType.new
data.extract_data

puts "\nExtraction tool for TSV file:"
data = TSVDataType.new
data.extract_data

File: DesignPatternsExercises/java/StrategyDesignPattern/Exercise/StrategyDesignPattern.java
public class StrategyDesignPattern {
    public static void main(String[] args) {
        System.out.println("Extraction tool for CSV file:");
        DataType dataType = new CSVDataType();
        dataType.extractData();

        System.out.println("\nExtraction tool for DB file:");
        dataType = new DBDataType();
        dataType.extractData();

        System.out.println("\nExtraction tool for TSV file:");
        dataType = new TSVDataType();
        dataType.extractData();
    }
}
